scaladoc-js/src/searchbar/engine/InkuireJSSearchEngine.scala

```diff
@@ -0,0 +1,50 @@
+package dotty.tools.scaladoc
+
+import scala.io.Source
+import dotty.tools.scaladoc.PageEntry
+import org.scalajs.dom.webworkers.Worker
+import org.scalajs.dom._
+import scala.scalajs.js.{ JSON, Dynamic }
+import scala.collection.mutable.ListBuffer
+import scala.scalajs.js
+import scala.scalajs.js.timers._
+import org.scalajs.dom.ext.Ajax
+import scala.scalajs.js.URIUtils
+
+class InkuireJSSearchEngine {
+
+  val scriptPath     = Globals.pathToRoot + "scripts/"
+  val worker: Worker = new Worker(scriptPath + "inkuire-worker.js")
+
+  def dynamicToPageEntry(d: Dynamic): PageEntry = {
+    PageEntry(
+      d.functionName.asInstanceOf[String],
+      d.prettifiedSignature.asInstanceOf[String],
+      d.pageLocation.asInstanceOf[String],
+      d.functionName.asInstanceOf[String],
+      "def",
+      List.empty
+    )
+  }
+
+  def query(s: String)(callback: PageEntry => Node)(endCallback: String => Node): List[PageEntry] = {
```

Comment:
I think the callbacks here should return Unit because engine should only know when to call it and shouldn't know what to do with return type (in this case Node is a type more related to frontend).

Reply:
Makes sense, done